File: Problems/Parking Lot/Java/floor/Floor.java
package floor;

import java.util.ArrayList;

import parkingSpot.ParkingSpot;

public class Floor {
    private String floorID;
    private ArrayList<ParkingSpot> parkingSpots;
    
    public Floor(String floorID, ArrayList<ParkingSpot> parkingSpots) {
        this.floorID = floorID;
        this.parkingSpots = parkingSpots;
    }

    public String getFloorID(){
        return floorID;
    }

    public ArrayList<ParkingSpot> getParkingSpots() {
        return parkingSpots;
    }
}

File: Problems/Parking Lot/Golang/floor/floor.go
package floor

import (
	"parking-lot/parkingspot"
)

// Floor represents a parking floor with multiple parking spots
type Floor struct {
	FloorID      string
	ParkingSpots []parkingspot.ParkingSpot
}

func NewFloor(floorID string, parkingSpots []parkingspot.ParkingSpot) *Floor {
	return &Floor{
		FloorID:      floorID,
		ParkingSpots: parkingSpots,
	}
}

func (f *Floor) GetFloorID() string {
	return f.FloorID
}

func (f *Floor) GetParkingSpots() []parkingspot.ParkingSpot {
	return f.ParkingSpots
}
